File: Processor/Source/Executable/Executable.py
from IExecutable import IExecutable 
from ResultWriter import ResultWriter
from Webpage import Webpage 

class Executable(IExecutable):
    def find_tag_by_css_selector(url, css_selector):
        webpage = Webpage(url)
        Executable.find_tag_by_css_selector_using_webpage(webpage,css_selector)

    def find_tag_by_css_selector2(url, page_iteration_selector, iteration_number, css_selector):
        return Executable.do_action_on_iteration(url,page_iteration_selector,iteration_number,"find_tag_by_css_selector",[css_selector])

    def find_tag_by_css_selector_using_webpage(webpage, css_selector):
        result = webpage.get_element_by_selector(css_selector).get_text()
        url = str(webpage.page_url)
        content = {
            "Url": url,
            "Content":result
        }
        result_writer = ResultWriter()
        result_writer.print_result(content)

    def get_page_numbers(url, page_iteration_selector):
        webpage = Webpage(url)
        return webpage.get_page_numbers(page_iteration_selector)

    def do_action_on_iteration(url, page_iteration_selector, iteration_number, action,data):
        webpage = Webpage(url)
        webpage.iterate_through_pages(iteration_number,page_iteration_selector)
        func = getattr(Executable, action+"_using_webpage")
        return func(webpage,*data)
    
    def get_iteration_links(url, iteration_selector):
        webpage = Webpage(url)
        return Executable.get_iteration_links_using_webpage(webpage,iteration_selector)
    

    def get_iteration_links2(url, page_iteration_selector, iteration_number, iteration_selector):
        return Executable.do_action_on_iteration(url,page_iteration_selector,iteration_number,"get_iteration_links",[iteration_selector])
    
    def get_iteration_links_using_webpage(webpage, iteration_selector):
        links = webpage.get_iteration_links(iteration_selector)
        return links
    

File: Processor/Source/PythonExecution/Executable.cs
using PythonInterfaceConverterLibrary;
using System.Collections.Generic;
using System.Reflection;

namespace PythonExecution
{
	public class Executable : IExecutable
	{
		public void find_tag_by_css_selector(string url, string css_selector)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(css_selector);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void find_tag_by_css_selector2(string url, string page_iteration_selector, string iteration_number, string css_selector)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(page_iteration_selector);
			parameters.Add(iteration_number);
			parameters.Add(css_selector);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void find_tag_by_content(string url, string text)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(text);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void find_tag_by_content2(string url, string page_iteration_selector, string iteration_number, string text)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(page_iteration_selector);
			parameters.Add(iteration_number);
			parameters.Add(text);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void count_texts_on_page(string url, string texts)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(texts);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void count_texts_on_page2(string url, string page_iteration_selector, string iteration_number, string texts)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(page_iteration_selector);
			parameters.Add(iteration_number);
			parameters.Add(texts);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void page_contains_text(string url, string text)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(text);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public void page_contains_text2(string url, string page_iteration_selector, string iteration_number, string text)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(page_iteration_selector);
			parameters.Add(iteration_number);
			parameters.Add(text);
			ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public string get_page_numbers(string url, string page_iteration_selector)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(page_iteration_selector);
			return ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public string get_iteration_links(string url, string iteration_selector)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(iteration_selector);
			return ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
		public string get_iteration_links2(string url, string page_iteration_selector, string iteration_number, string iteration_selector)
		{
			string methodName = MethodInfo.GetCurrentMethod().Name;
			List<object> parameters = new List<object>();
			parameters.Add(url);
			parameters.Add(page_iteration_selector);
			parameters.Add(iteration_number);
			parameters.Add(iteration_selector);
			return ApiInvocation.RunCommandOnPython(methodName, parameters);
		}
	}
}